packages/enzyme/src/ShallowWrapper.js

```diff
@@ -1033,6 +1033,34 @@ class ShallowWrapper {
     return this.props()[propName];
   }
 
+  /**
+   * Returns a wrapper of the node rendered by the provided render prop.
+   *
+   * @param {String} propName
+   * @returns {ShallowWrapper}
+   */
+  renderProp(propName, ...args) {
+    const Wrapper = props => props.children;
+    return this.single('renderProp', () => {
+      if (typeof propName !== 'string') {
+        throw new TypeError('`propName` must be a string');
+      }
+      const prop = this.prop(propName);
+      if (!prop) {
```

Comment:
`prop` could be falsy with `0`, `false`, and a number of other options. Perhaps instead this should do `const props = this.props()`, and then do `has(props, propName)`?

Reply:
that's a reasonable alternative; altho it's nice to be able to give a better error message "this prop doesn't exist" versus "this prop isn't a function".